packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -188,12 +188,21 @@ class CupertinoPageRoute<T> extends PageRoute<T> {
 
   /// True if a Cupertino pop gesture is currently underway for [route].
   ///
+  /// This returns true for both the top route and the bottom route of a
+  /// pop gesture.
+  ///
   /// See also:
   ///
   ///  * [popGestureEnabled], which returns true if a user-triggered pop gesture
   ///    would be allowed.
-  static bool isPopGestureInProgress(PageRoute<dynamic> route) => _popGestureInProgress.contains(route);
-  static final Set<PageRoute<dynamic>> _popGestureInProgress = Set<PageRoute<dynamic>>();
+  static bool isPopGestureInProgress(PageRoute<dynamic> route) {
+    return _popGestureInProgress.contains(_animationControllerForTransitionAnimation(route.animation))
+        || _popGestureInProgress.contains(_animationControllerForTransitionAnimation(route.secondaryAnimation));
```

Comment:
I'm not sure why _animationControllerForTransitionAnimation will always return route.secondaryAnimation if a route is using it. It seems like it would in simple cases, but all cases?

Reply:
Took a different approach